packages/runtime-common/schema-analysis-plugin.ts

```diff
@@ -15,238 +10,306 @@ export interface ExternalReference {
 
 export interface InternalReference {
   type: 'internal';
-  classIndex?: number;
+  classIndex?: number; //the internal index only applies to possibleCardsOrFields array
 }
 
 export type ClassReference = ExternalReference | InternalReference;
 
-export type BaseDeclaration = {
+export interface BaseDeclaration {
   localName: string | undefined;
   exportedAs: string | undefined;
   path: NodePath;
-  type: 'class' | 'function';
-};
+}
 
-export interface PossibleCardOrFieldClass extends BaseDeclaration {
+export interface FunctionDeclaration extends BaseDeclaration {
+  type: 'function';
+  path: NodePath<t.FunctionDeclaration>;
+}
+
+export interface ClassDeclaration extends BaseDeclaration {
+  type: 'class';
+  path: NodePath<t.ClassDeclaration>;
+}
+
+export interface PossibleCardOrFieldDeclaration extends BaseDeclaration {
+  type: 'possibleCardOrField';
   super?: ClassReference; // this is optional to allow to be inclusive of base def class
   possibleFields: Map<string, PossibleField>;
   path: NodePath<t.ClassDeclaration>;
 }
 
+export interface Reexport extends BaseDeclaration {
+  type: 'reexport';
+}
+
 export interface PossibleField {
   card: ClassReference;
   type: ExternalReference;
   decorator: ExternalReference;
   path: NodePath<t.ClassProperty>;
 }
 
-// a module declaration should be (an item of focus within a module)
-// - exported function or class
-// - exported card or field
-// - unexported card or field
-export type Declaration = PossibleCardOrFieldClass | BaseDeclaration;
+export type Declaration =
+  | PossibleCardOrFieldDeclaration
+  | FunctionDeclaration
+  | ClassDeclaration
+  | Reexport;
 
 export interface Options {
-  possibleCardsOrFields: PossibleCardOrFieldClass[]; //cards may not be exports
+  possibleCardsOrFields: PossibleCardOrFieldDeclaration[]; //cards may not be exports
   declarations: Declaration[];
 }
 
+interface State {
+  opts: Options;
+  insideCard: boolean;
+}
+
 export function schemaAnalysisPlugin(_babel: typeof Babel) {
   return {
-    visitor: {
-      FunctionDeclaration: {
-        enter(path: NodePath<t.FunctionDeclaration>, state: State) {
-          let localName = path.node.id ? path.node.id.name : undefined;
-          if (t.isExportDeclaration(path.parentPath)) {
-            // == handle direct export ==
-            state.opts.declarations.push({
+    visitor: visitors.merge([coreVisitor, reExportVisitor]),
```

Comment:
I split into 2 visitors

- core visitor looks for declaration and code 
- reExport visitor looks at the export declarations and specifiers